packages/pwa-kit-dev/src/configs/webpack/overrides-plugin.test.js

```diff
@@ -216,13 +219,193 @@ describe('overrides plugin', () => {
         expect(resolver.doResolve).not.toHaveBeenCalledWith()
     })
 
-    test('jsx base template files can be replaced by tsx files', () => {})
+    test('jsx base template files can be replaced by tsx files', () => {
+        const REQUEST_PATH = 'newExtension'
+        const REQUEST_BASE_EXTENSION = '.jsx'
+        const REQUEST_OVERRIDE_EXTENSION = '.tsx'
 
-    test('npm @namespaces resolve correctly', () => {})
+        const testRequestContext = {
+            _ResolverCachePluginCacheMiss: true,
+            context: {
+                issuer: path.join('./', 'fake-file.js')
+            },
+            path: path.join(
+                './',
+                'node_modules',
+                EXTENDS_TARGET,
+                REQUEST_PATH,
+                REQUEST_BASE_EXTENSION
+            ),
+            request: `${EXTENDS_TARGET}/${REQUEST_PATH}`
+        }
+
+        let {resolver, callback} = setupResolverAndCallback(
+            null,
+            testRequestContext,
+            null,
+            {},
+            callback
+        )
+
+        const overridesResolver = new OverridesResolverPlugin(options)
+        overridesResolver.handleHook(testRequestContext, {}, callback, resolver)
+
+        expect(callback).toHaveBeenCalled()
+        expect(resolver.ensureHook).toHaveBeenCalled()
+        expect(resolver.doResolve).toHaveBeenCalledWith(
+            null,
+            {
+                _ResolverCachePluginCacheMiss: true,
+                context: {
+                    issuer: path.join('./', 'fake-file.js')
+                },
+                path: path.join(REWRITE_DIR, REQUEST_PATH + REQUEST_OVERRIDE_EXTENSION),
+                request: `${EXTENDS_TARGET}/${REQUEST_PATH}`
+            },
+            expect.anything(),
+            expect.anything(),
+            expect.anything()
+        )
+    })
+
+    test('overridesDir and projectDir are normalized with leading slash and forward slashes', () => {
```

Comment:
This test is a bit if a catch-all for windows paths. Are there other windows specific situations we want to look out for?